rust/candid/src/de.rs

```diff
@@ -337,16 +334,13 @@ impl<'de> Deserializer<'de> {
         }
         Ok(())
     }
-    fn is_zero_sized_type(&self, t: &Type) -> bool {
```

Comment:
Conceptually, we should check this condition for every (nested) value and limit all such (nested) zero-sized values at 2M in total.

Reply:
See above. That would be overcounting, e.g., `record { null; null }` would be counted as 3 instead of 2.